posthog/utils.py

```diff
@@ -326,6 +333,19 @@ def render_template(template_name: str, request: HttpRequest, context: Dict = {}
 
     context["posthog_app_context"] = json.dumps(posthog_app_context, default=json_uuid_convert)
 
+    if not posthog_distinct_id:
+        posthog_distinct_id = str(uuid.uuid4())
+
+    feature_flags = posthoganalytics.get_all_flags(posthog_distinct_id, only_evaluate_locally=True)
```

Comment:
I assume this line won't introduce 300ms - 3s delays?

Reply:
Exactly, it's at worst a 20ms lag, because of the `only_evaluate_locally` param